File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    
    max_sizes = 0
    for i in range(len(sizes)):
        max_sizes = max(max(sizes[i]), max_sizes)
    
    min_sizes = 0
    for i in range(len(sizes)):
        min_sizes = max(min(sizes[i]), min_sizes) 
    
    answer = min_sizes * max_sizes
    
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int max_width = 0;
    int max_height = 0;
    
    for(const auto& v : sizes) {
        int width = max(v[0], v[1]);
        int height = min(v[0], v[1]);
        max_width = max(max_width, width);
        max_height = max(max_height, height);
    }
    
    return max_width * max_height;
}